lend/src/hooks/referralContext.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { ReactElement, useCallback, useEffect, useState } from "react";
+import {
+  AccountSummary,
+  createOrReturnCode,
+  getAccountSummary,
+} from "../utils/referrals";
+import useWeb3Wallet from "./useWeb3Wallet";
+
+export type ReferralContextType = {
+  referralAccountSummary: AccountSummary;
+  referralCode: string;
+  referralLoading: boolean;
+};
+
+const defaultReferralAccountSummary: AccountSummary = {
+  address: "---",
+  totalReferralRewards: 0,
+  userReferred: 0,
+  referralCodeUsed: "---",
+};
+
+export const defaultReferralData: ReferralContextType = {
+  referralAccountSummary: defaultReferralAccountSummary,
+  referralCode: "",
+  referralLoading: true,
+};
+
+export const ReferralContext =
+  React.createContext<ReferralContextType>(defaultReferralData);
+
+export const ReferralContextProvider: React.FC<{
+  children: ReactElement;
+}> = ({ children }) => {
+  const [referralAccountSummary, setAccountSummary] = useState(
+    defaultReferralAccountSummary
+  );
+  const [referralCode, setReferralCode] = useState("");
+  const [referralLoading, setReferralLoading] = useState(true);
+  const { account } = useWeb3Wallet();
+
+  const getAndSetAccountSummary = useCallback(async () => {
+    if (account) {
+      const data = await getAccountSummary(account);
+      await new Promise((resolve) => setTimeout(resolve, 3000));
```

Comment:
whats the reason for the timeout?

Reply:
Removed in [da914ae](https://github.com/ribbon-finance/ribbon-frontend/pull/624/commits/da914ae3a10539aa9d119085508d60ebfdf41c86), mistakenly left from testing. Thanks for catching!